### ISSUE TITLE
getroute doesn't account for spendable_msatoshi

### ISSUE THREAD

**Issue by s-tikhomirov** [ElementsProject/lightning#3526]
### Issue and Steps to Reproduce

`getroute` suggests routes that are guaranteed to fail. Consider a local channel with the following properties:
```
"msatoshi_to_us": 4500000000,
"spendable_msatoshi": 3817334952,
"htlcs": [
  {
    "direction": "out",
    "msatoshi": 637395048,
  }
]
```

`getroute` routes through this channel for an amount of 4 Msat, which is lower than `spendable_msatoshi` (another channel of mine is excluded to force `getroute` to go through the channel in question):

```
$ lightning-cli getroute -k id=038863cf8ab91046230f561cd5b386cbff8309fa02e3f0c3ed161a3aeb64a643b9 riskfactor=1 exclude='["1665882x256x1/0"]' msatoshi=4000000000
{
   "route": [
      {
         "id": "0277622bf4c497475960bf91bd3c673a4cb4e9b589cebfde9700c197b3989cc1b8",
         "channel": "1665706x160x1",
         "direction": 0,
         "msatoshi": 4000005000,
         "amount_msat": "4000005000msat",
         "delay": 153,
         "style": "legacy"
      },
      {
         "id": "038863cf8ab91046230f561cd5b386cbff8309fa02e3f0c3ed161a3aeb64a643b9",
         "channel": "1571601x34x0",
         "direction": 0,
         "msatoshi": 4000000000,
         "amount_msat": "4000000000msat",
         "delay": 9,
         "style": "tlv"
      }
   ]
}

```

### `getinfo` output
```
{
   "id": "026e7cb4237b4042db75a991af29c2d7704780c8e7fbebe5fdbad8f38e958dfba5",
   "alias": "clightningM",
   "color": "026e7c",
   "num_peers": 6,
   "num_pending_channels": 0,
   "num_active_channels": 3,
   "num_inactive_channels": 4,
   "address": [],
   "binding": [
      {
         "type": "ipv4",
         "address": "127.0.0.1",
         "port": 9737
      }
   ],
   "version": "v0.8.0-40-g899f5de",
   "blockheight": 1665944,
   "network": "testnet",
   "msatoshi_fees_collected": 0,
   "fees_collected_msat": "0msat",
   "lightning-dir": "/home/sergei/.lightning/testnet"
}
```

**Comment by ZmnSCPxj** [comment]
This was done deliberately, as only local channels have this information and remote channels do not.  `pay` plugin, for example, inspects the local channels `spendable_msatoshi` in a separate pass before calling `getroute`. https://github.com/ElementsProject/lightning/blob/0b123a3cec0222718361ad931e555c8de881035e/plugins/pay.c#L1101-L1107

This was done to reduce inter-process communication with `gossipd`; we only hand over actual inter-node gossip, not our local state.

In principle, we could reorganize things so current `getroute` is `getroute_raw`, then add a `excludelowchannels` command that creates an `excludes` list of channels whose `spendable_msatoshi` is below a target price (+ `maxfeepercent`).  Then a new `getroute` would just call `excludelowchannels` and then call `getroute_raw`.  Same can be done with shadow routing as well.  This helps if people want to re-implement `pay`, which I argue is the entire point of moving it into a plugin anyway.

Basically: it is a deliberate architectural decision that `getroute` ignores local channel state.  Whether to call it `getroute` or `getroute_raw`, that is the question.

**Comment by s-tikhomirov** [comment]
Thanks a lot for the clarification!

Am I correct that if I call `getroute` and then `sendpay` with the route which does not actually have sufficient capacity, I'll get `WIRE_REQUIRED_CHANNEL_FEATURE_MISSING`? This is what I see in my logs, and it looks counter-intuitive... Though I'm not 100% sure those errors are caused by this issue.

**Comment by ZmnSCPxj** [comment]
It should return `WIRE_TEMPORARY_CHANNEL_FAILURE`.  You should get `WIRE_REQUIRED_CHANNEL_FEATURE_MISSING` only if you are trying to send > 42.94967295 msatoshi.

**Comment by sumBTC** [comment]
@s-tikhomirov I created a bash [shell script](https://BitcoinDayTrader.nl/getroute_mine.sh) that makes `getroute` fail it there isn't enough liquidity in the channels of the known neighbour nodes. Maybe it's useful to you. 

@ZmnSCPxj A **real** `getroute` and `getroute_raw`, as you suggested would be useful

**Comment by sangbida** [comment]
This issue is being closed as part of issue cleanup.

**Reason**: Very old issue (6.0 years old) no recent activity

If you believe this issue should remain open, please comment and we can reopen it.
